snitchpy/__init__.py

```diff
@@ -589,14 +618,34 @@ async def call():
                         pass
                     elif command == "tail":
                         self._tail_request(cmd)
+                    elif command == "kv":
+                        self._handle_kv(cmd)
                     else:
                         self.log.error(f"Unknown response type: {cmd}")
                 except ValueError as e:
                     self.log.error(f"Received invalid command: {e}")
 
         self.log.debug("Starting register looper")
         asyncio.set_event_loop(self.register_loop)
-        self.register_loop.run_until_complete(call())
+
+        while not self.exit.is_set():
+            try:
+                self.register_loop.run_until_complete(call())
+            except Exception as e:
+                self.log.debug(
+                    f"Register looper lost connection, retrying in {DEFAULT_GRPC_RECONNECT_INTERVAL}s..."
+                )
+                try:
+                    time.sleep(DEFAULT_GRPC_RECONNECT_INTERVAL)
+                    self.register_channel = Channel(
+                        host=self.host, port=self.port, loop=self.register_loop
+                    )
+                    self.grpc_channel = Channel(
+                        host=self.host, port=self.port, loop=self.grpc_loop
+                    )
+                    self._add_audiences()
+                except Exception as e:
+                    self.log.error(f"reconnection failed: {e}")
```

Comment:
Ah there's already a logger :) Is it injectable into the lib?

Reply:
It's python's default logger that's built into the stdlib. It works like a singleton and you can configure it outside the scope of the library